.github/copilot-instructions.md

```diff
@@ -0,0 +1,95 @@
+
+#### Prerequisites for New Convention Areas
+- **Domain Expertise**: Must have a group of people familiar with the domain
+- **Active Involvement**: Contributors must be involved with instrumentation efforts in that area
+- **Long-term Commitment**: Must have committed maintainers as points of contact for pull requests, issues, questions, and ongoing maintenance
+
+#### Process for New Areas
+1. **Project Management**: Follow [OpenTelemetry project management](https://github.com/open-telemetry/community/blob/main/project-management.md) guidelines
+2. **Codeowners**: New areas MUST have assigned codeowners
+
+## General Review Principles
+- **YAML Models**: All attributes, metrics, and conventions are formally defined in YAML files under `model/` directory
+  - Validate against [schema](https://github.com/open-telemetry/weaver/blob/main/schemas/semconv.schema.json)
+  - Verify clear `brief` and `note` sections with realistic examples
+  - Confirm proper RFC 2119 keywords (MUST, SHOULD, MAY, etc.)
+
+- **Scope**: ensure contributions belong here:
+  - **SHOULD be included:**
+    - Instrumentations hosted in OpenTelemetry
+    - Common attributes and conventions used across multiple instrumentations
+    - Conventions that benefit from centralized policy enforcement
+
+  - **SHOULD NOT be included:**
+    - Instrumentations following external schemas not fully compatible with OpenTelemetry
+    - Domain-specific conventions better maintained in their own repositories
+
+### Naming Conventions
+- Use lowercase with dot-separated namespaces (e.g., `service.name`)
+- Multi-word components use snake_case (e.g., `http.response.status_code`)
+- Only widely recognized abbreviations (HTTP, CPU, AWS, etc.)
+- Proper namespace hierarchy, avoid name collisions
+- No name reuse within categories
+
+## Attribute Review Guidelines
+
+### Requirements
+- Clear benefit to end users with usage in spans, metrics, events, or entities
+- Follow stability progression (development → stable)
+- Reuse existing attributes when possible, check existing namespaces first
+- Use appropriate type (string, int, double, boolean, array, enum)
+- Define enums for closed sets, avoid unbounded values (>1KB strings, >1000 elements)
+- Mark PII/sensitive data with warnings
+
+### Types and Usage
+- **Timestamps**: ISO 8601 string format
+- **Arrays**: Homogeneous; separate attributes for different concepts
+- **Complex values**: Flatten when possible
+- **Template attributes**: Dynamic names (last segment only)
+
+## Event Conventions
+- MUST have unique, low-cardinality event name
+- SHOULD have attributes for context, SHOULD NOT use body
+
+## Error Handling Conventions
+- **Spans**: Set status to Error, populate `error.type`, set description when helpful
+- **Metrics**: Include `error.type` attribute for filtering and analysis
+- **Exceptions**: Record as span events or log records using SDK APIs
```

Comment:
```suggestion
- **Exceptions**: Record as exception events
```

Reply:
good catch! let's remove this line - conventions don't really include exception attributes anyway.